test/ValidityTest.hs

```diff
@@ -152,7 +152,7 @@ extensionsSpec = do
          forAllShrink genValid shrinkValidRelFile $ \file ->
              forAllShrink genValid shrinkValidExtension $ \(Extension ext) ->
                 (toFilePath . fromJust . addFileExtension ext) file
-                 `shouldBe` toFilePath file ++ ext
+                 == toFilePath file ++ ext
```

Comment:
`shouldBe` was _not_ redundant here.
Now your error messages will be unhelpful.